custom_components/tahoma/__init__.py

```diff
@@ -198,3 +229,6 @@ async def update_listener(hass: HomeAssistant, entry: ConfigEntry):
         coordinator.original_update_interval = new_update_interval
 
         await coordinator.async_refresh()
+
+    if entry.options[CONF_REFRESH_STATE_INTERVAL]:
+        print("Change refresh state of async_track_time_interval")
```

Comment:
@tetienne would you have a clue how to solve this? `async_track_time_interval` returns an unsub function, however I don't think I can create a new task calling `handle_refresh_states()` from this function...

Reply:
I'm sorry, but I have no idea what async_track_time_interval is :(

Reply:
It is more related to Python and scoping in general. We currently have a lot of functions in our `async_setup_entry`, which is very cluttered. 

In `update_listener` I need to create a new task calling the `handle_refresh_states` function, which I can not seem to access there. I solved it with some duplicate code, but I would love to have this all a bit more clean. 
(at least it works now).